queries/futures/useGetFuturesPositionForAccount.ts

```diff
@@ -1,54 +1,108 @@
 import { NetworkId } from '@synthetixio/contracts-interface';
-import request, { gql } from 'graphql-request';
 import { useQuery, UseQueryOptions } from 'react-query';
-import { useRecoilValue } from 'recoil';
+import { useSetRecoilState } from 'recoil';
 
 import QUERY_KEYS from 'constants/queryKeys';
 import Connector from 'containers/Connector';
-import { futuresAccountTypeState, selectedFuturesAddressState } from 'store/futures';
+import { positionHistoryState } from 'store/futures';
 import logError from 'utils/logError';
 
-import { FUTURES_POSITION_FRAGMENT } from './constants';
-import { PositionHistory } from './types';
-import { getFuturesEndpoint, mapTradeHistory } from './utils';
+import { getFuturesPositions } from './subgraph';
+import { FuturesAccountTypes, PositionHistory, PositionHistoryState } from './types';
+import { getFuturesEndpoint, mapFuturesPositions } from './utils';
 
-const useGetFuturesPositionForAccount = (options?: UseQueryOptions<any>) => {
+const DEFAULT_POSITION_HISTORY: PositionHistoryState = {
+	[FuturesAccountTypes.CROSS_MARGIN]: [],
+	[FuturesAccountTypes.ISOLATED_MARGIN]: [],
+};
+
+const useGetFuturesPositionForAccount = (
+	account?: string,
```

Comment:
Having this optional account and conditional state setting feels a bit off, I think ideally we avoid setting state in this query and have a higher level wrapper query that calls this with users wallet and sets positionHistory state.

Reply:
Fixed this by splitting into two queries. One is a regular `react-query` which fetches the data for a specified account. The other calls that query, and has a `useEffect` to set the state value when the data changes.